liquidhaskell-boot/src/Language/Haskell/Liquid/Constraint/RewriteCase.hs

```diff
@@ -0,0 +1,138 @@
+module Language.Haskell.Liquid.Constraint.RewriteCase 
+    (getCaseRewrites) 
+    where
+
+import           Language.Fixpoint.Types
+import qualified Language.Fixpoint.Misc as M
+import           Language.Haskell.Liquid.Constraint.Types
+import           Language.Haskell.Liquid.Types.Types
+import           Language.Haskell.Liquid.Types.RType
+
+import           Data.Maybe
+import           Data.Tuple
+
+import qualified Data.HashMap.Strict as M
+import qualified Data.HashSet        as S
+
+getCaseRewrites :: CGEnv -> SpecType -> LocalRewrites
+getCaseRewrites γ spec =
+    let Reft (_, refinement) = ur_reft $ rt_reft spec
+        -- Names of constustors both type constructors and data constructors
+        ctors   = toSet $ seBinds  $ constEnv γ
+        -- All the global names (top level functions, types, etc...)
+        globals = toSet $ reGlobal $ renv     γ
+    in unloop
+       $ concatMap (uncurry $ unify ctors globals)
+       $ groupUnifiableEqualities 
+       $ getEqualities refinement
```

Comment:
I'm realizing that if the pipeline is fed a single equality `f = \x -> x` with `f` local, it is not going to generate a rewrite for `f`. While I realized this is not the purpose of `getCaseRewrites`, I'm left wondering how to make a rewrite for it. Is this future work?

Reply:
We could add that case is not going to create any problem in principle, it's the same thing as get equality being right biased it was thought as a way to extract equalities when performing dependant pattern matching on dependant inductive indexed types and in LH we generate constraint for those in a standard format (`prop e = T e1 ... en`) but I believe that in spirit of making the code more general as we did by removing the right bias we could also handle those cases